### PR TITLE
Document how to use testcontainers service connections for test slices

### PR DESCRIPTION
**Pull request by quaff**
Close GH-35121

### CODE REVIEW

File: spring-boot-project/spring-boot-docs/src/docs/asciidoc/features/testing.adoc [hunk 1]
```diff
@@ -938,6 +938,9 @@ This is done by automatically defining a `Neo4jConnectionDetails` bean which is
 
 NOTE: You'll need to add the `spring-boot-testcontainers` module as a test dependency in order to use service connections with Testcontainers.
 
+NOTE: You'll need to add `@ImportAutoConfiguration(ServiceConnectionAutoConfiguration.class)` if you are using <<features#features.testing.spring-boot-applications.autoconfigured-tests, test slices>>.
```
Comment (eddumelendez) [inline spring-boot-project/spring-boot-docs/src/docs/asciidoc/features/testing.adoc:941]: I've been using `@ServiceConnection` along with `@JdbcTest` without adding this. See https://github.com/eddumelendez/testcontainers-samples/blob/spring-boot-3.1.0-RC1/spring-boot-cockroachdb-flyway/src/test/java/com/example/springbootcockroachdbflyway/CockroachDbTest.java

Comment (quaff) [reply spring-boot-project/spring-boot-docs/src/docs/asciidoc/features/testing.adoc:941]: @eddumelendez If you define it as `@Bean` managed by spring not `@Container` field managed by testcontainers, then you need `@ImportAutoConfiguration(ServiceConnectionAutoConfiguration.class)`.

Comment (quaff) [reply spring-boot-project/spring-boot-docs/src/docs/asciidoc/features/testing.adoc:941]: I've amend the commit.

Comment (eddumelendez) [reply spring-boot-project/spring-boot-docs/src/docs/asciidoc/features/testing.adoc]: Just out of curiosity why would you do that given

```
@Container
@ServiceConnection
static final GenericContainer<?> ...
```

would already work for your existing tests, without adding any additional configuration.

Comment (quaff) [reply spring-boot-project/spring-boot-docs/src/docs/asciidoc/features/testing.adoc]: I'd like to let spring manage it's lifecycle if possible, I'm doing this before `spring-boot-testcontainers` existed.

```java
abstract class JdbcDatabase<T extends JdbcDatabaseContainer<?>> {

	private final Class<T> containerClass;

	@SuppressWarnings("unchecked")
	JdbcDatabase() {
		this.containerClass = (Class<T>) ResolvableType.forClass(getClass()).as(JdbcDatabase.class).resolveGeneric(0);
	}

	protected String getImageName() {
		return getClass().getSimpleName().toLowerCase();
	}

	protected String getImageTag() {
		return "latest";
	}

	protected String getImage() {
		return getImageName() + ':' + getImageTag();
	}

	@Bean(initMethod = "start")
	T container() throws Exception {
		return this.containerClass.getConstructor(String.class).newInstance(getImage());
	}

	@Bean
	DataSource dataSource(T container) {
		HikariDataSource ds = new HikariDataSource();
		ds.setJdbcUrl(container.getJdbcUrl());
		ds.setUsername(container.getUsername());
		ds.setPassword(container.getPassword());
		return ds;
	}

}

@TestConfiguration(proxyBeanMethods = false)
public class MySQL extends JdbcDatabase<MySQLContainer<?>> {

}

```

**Comment by wilkinsona** [comment]
Thanks for the proposal. We'd prefer to try to fix the underlying problems. I've opened #35252 and #35253.

**Pull request closed without merging**